vllm/entrypoints/sync_openai/api_server.py

```diff
@@ -173,8 +174,9 @@ async def _check_model(request: Union[CompletionRequest,
 
 async def _guided_decode_logits_processor(request, tokenizer):
     decoding_config = runner.engine_config.decoding_config
-    guided_decoding_backend = request.guided_decoding_backend \
-            or decoding_config.guided_decoding_backend
+    assert decoding_config is not None
```

Comment:
Strictly speaking not required if it arrives in the request

Reply:
Yeah oops the logic for the assert as written currently isn't right...

For context, this was done because the lint complained that `decoding_config` could be `None`.
Perfunctory read seems to indicate that that's not possible because the default initialization prevents it. Unfortunately the linter can't trace types that far.

So either we do a `# type: ignore` or do this assert only if `request.guided_decoding_backend` is a dud.